code/renderers/react/template/components/Form.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+export const Form = ({ onSuccess }) => {
+  const [value, setValue] = useState('');
+  const [succeed, setSucceed] = useState(false);
+  const [error, setError] = useState(null);
+
+  function onSubmit(event) {
+    event.preventDefault();
+    if (succeed) {
+      setError(null);
+      onSuccess(value);
+    } else {
+      setTimeout(() => {
+        setError(`Submitted '${value}' when not allowed!`);
+      }, 1000);
+    }
+  }
+
+  return (
+    <form id="interaction-test-form" onSubmit={onSubmit}>
+      <style>{`
+        #interaction-test-form button:hover {
+          background: red;
+        }
+      `}</style>
```

Comment:
I think this pattern will result in a lot of duplicated code, code that is actually not renderer-specific (CSS) and so this is not the direction I think we should take.

Let's find some other way. I suggest a static stylesheet, that's shared.

Reply:
@shilman and I decided let's just drop the styles completely.